include/aws/iotdevice/private/secure_tunneling_operations.h

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#ifndef AWS_IOTDEVICE_SECURE_TUNNELING_OPERATION_H
+#define AWS_IOTDEVICE_SECURE_TUNNELING_OPERATION_H
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/linked_list.h>
+#include <aws/common/ref_count.h>
+#include <aws/iotdevice/private/serializer.h>
+#include <aws/iotdevice/secure_tunneling.h>
+
+/*********************************************************************************************************************
+ * Operations
+ ********************************************************************************************************************/
+
+struct aws_secure_tunnel_operation;
+
+enum aws_secure_tunnel_operation_type {
+    AWS_STOT_NONE,
+    AWS_STOT_PING,
+    AWS_STOT_DATA,
+    AWS_STOT_STREAM_RESET,
+    AWS_STOT_STREAM_START
+};
+
+struct aws_secure_tunnel_message_storage {
+    struct aws_allocator *allocator;
+    struct aws_secure_tunnel_message_view storage_view;
+
+    bool ignorable;
```

Comment:
Storage is used only for outbound messages and we don't send anything that sets this field to true (false being the default) since we don't send anything that has an unexpected type.